### PR TITLE
Feature/mouse unit tests

### PR DESCRIPTION
**Pull request by AlexandreSanscartier**
Created unit tests for the Mouse.js class.

### CODE REVIEW

File: src/core/errorHandling/errors/UnhandledEventError.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { UnhandledEventErrorText } from '../ErrorTexts';
+
+/**
+* Defines the class for the UnhandledEventError
+* TODO: rephrase this Should be thrown in every method that needs to be overriden
+*/
+class UnhandledEventError extends Error {
```
Comment (Zargath) [inline src/core/errorHandling/errors/UnhandledEventError.js]: I think this should be "UnhandledHtmlEventError"
https://www.w3schools.com/tags/ref_eventattributes.asp

File: src/core/errorHandling/errors/UnhandledEventError.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { UnhandledEventErrorText } from '../ErrorTexts';
+
+/**
+* Defines the class for the UnhandledEventError
+* TODO: rephrase this Should be thrown in every method that needs to be overriden
```
Comment (Zargath) [inline src/core/errorHandling/errors/UnhandledEventError.js]:
```suggestion
* TODO: rephrase this Should be thrown when an html event is not handled by the system.
```

File: tests/Mouse.test.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
+import UnhandledEventError from '../src/core/errorHandling/errors/UnhandledEventError';
+import InvalidArguementError from '../src/core/errorHandling/errors/InvalidArguementError';
+import Mouse from '../src/inputEngine/Mouse';
+
+describe('buttonDownOnce', () => {
+  it('should be true when mouse button is down once', () => {
```
Comment (Zargath) [inline tests/Mouse.test.js]: I think we should try to stick to Simple or PO like language in our tests

Comment (AlexandreSanscartier) [reply tests/Mouse.test.js]: done!

File: src/inputEngine/Mouse.js [hunk 4]
```diff
@@ -56,60 +64,59 @@ class Mouse {
    * @param {MouseEvent} e The MouseEvent
    */
   mouseMove(e) {
-    if (e.layerX || e.layerX === 0) { // FireFox
-      this.dx = e.layerX;
-      this.dy = e.layerY;
-    } else if (e.offsetX || e.offsetX === 0) { // Opera
-      this.dx = e.offsetX;
-      this.dy = e.offsetY;
+    if (e.clientX || e.clientX === 0) { // FireFox
```
Comment (Zargath) [inline src/inputEngine/Mouse.js]:
```suggestion
    if (e.clientX || e.clientX === 0) {
```

**Pull request merged**